### ISSUE TITLE
Is there any configuration that can dynamically modify the config.json in the guest vm?

### ISSUE THREAD

**Issue by cnbailian** [kata-containers/kata-containers#5606]
Like `precreate` hook for podman

My usage scenario:

**Comment by Ji-Xinyou** [comment]
Kata also supports guest hooks, but it requires pre-modification in guest kernel's rootfs.

**Comment by cnbailian** [comment]
@Ji-Xinyou 
I have tried using the `guest_hook_path` configuration, but the config.json modified by the hooks does not affect the container.

**Comment by Ji-Xinyou** [comment]
> @Ji-Xinyou I have tried using the `guest_hook_path` configuration, but the config.json modified by the hooks does not affect the container.

Is the `config.json` you referred to the oci spec that containerd passed down? 
If so, it is first deserialized in runtime, and passed to agent directly. I don't think the `config.json` is in guest.
However, if you want to share states between containers through a directory, you can you `volume`

**Comment by cnbailian** [comment]
@Ji-Xinyou 
No, it is created by the agent with the contents of containers oci spec in guest. 
What I want is to be able to affect the creation of the container by modifying this configuration.

I don't want to share states, I just want to mount the directory of guest, not host.

**Comment by liubin** [comment]
@cnbailian interesting case.

Currently, only shim/agent can modify OCI spec/config.json now, hooks can't change the config.json.

There are two ways to implement your case:

- support `precreate` to let hooks modify the config.json
- let users configure a list of guest mount directories and add them to the `.mounts` for the new containers.

The second one is easier and safer than the first.

But I'm eager to know how many cases need this feature.
